src/httpAuthScheme.ts

```diff
@@ -370,6 +370,75 @@ export class Token {
         return token;
     }
 
+    static parse(tokenTypeEntry: TokenTypeEntry, data: string): Token {
+        // Consumes data:
+        //   challenge="abc...", token-key="123..."
+
+        const attributes = data.split(',');
+        let ppToken = undefined;
+
+        for (const attr of attributes) {
+            const idx = attr.indexOf('=');
+            let attrKey = attr.substring(0, idx);
+            let attrValue = attr.substring(idx + 1);
+            attrValue = attrValue.replaceAll('"', '');
+            attrKey = attrKey.trim();
+            attrValue = attrValue.trim();
+
+            switch (attrKey) {
+                case 'token': {
+                    const tokenEnc = base64url.parse(attrValue);
+                    ppToken = Token.deserialize(tokenTypeEntry, tokenEnc);
+                    break;
```

Comment:
breaks the switch-case or breaks the for-loop?
returns always the first one?

Reply:
I removed the switch, but kept the for loop. WWW-Authenticate header can contain more than one attribute, even if not specified by the protocol.